contributor-docs/operations/tests/test-implementation-guide.md

```diff
@@ -0,0 +1,147 @@
+---
+title: SCS Conformance Test Implementation Guide
+type:
+status: Draft
+track: Global
+---
+
+SovereignCloudStack (SCS) uses [conformance tests][gh-scripts] to certify
+compliance of a given cloud offering with respect to a given [certificate
+scope][cert-scopes] such as *SCS Compatible IaaS v4*.
+Our aim is that these tests are reliable, consistent and comprehensible for
+the users.
+This document is a guideline for conformance test authors and summarizes the
+current best practices to achieve this goal.
+
+## 1. Mapping of RFC2119 Keywords
+
+Test authors working on new and existing conformance tests for an SCS standard
+must implement the keywords such as MUST and SHOULD according to the [SCS
+RFC2119 Keyword Test Guide][scs-rfc2119-guide].
+
+## 2. Unit and Regression Tests
+
+Test authors are *strongly* encouraged to include unit and regression tests for
+the conformance test's logic.
+Conformance tests will inevitably contain some non-trivial algorithms, be it for
+parsing flavor names or evaluating CVE vulnerability reports.
+As a bonus, a well written unit test also makes it easier for reviewers to
+cross check a given pull request with new or enhanced conformance tests.
+
+In general, unit and regression tests for the conformance tests are located in
+the same location as the conformance tests, that is, the `Tests/` directory of
+the [SCS standards repository][scs-standards].
+Setup and development of the unit tests is described in the [SCS conformance
+tests README][tests-readme].
+
+### Naming Conventions
+
+Any module that ends in `_test` will be picked up as a unit test module by
+pytest in our CI pipeline (in other words, all files matched by the pattern
+`Tests/**/*_test.py`).
+Pytest will execute any function prefixed with `test_` as a test function, or,
+alternatively, tests based on the `unittest` module from the Python standard
+library.
+
+Follow these naming rules:
+
+* The name of a unit test module is constructed from the module to be unit
+  tested and the `_test` suffix. For example, unit tests for `flavor_names.py`
+  should reside in `flavor_names_test.py`.
+* The name of a Python module in general should be a valid [Python
+  identifier][python-identifiers], to allow imports via the `import` statement.
+  For example, use `flavor_names.py` instead of `flavor-names.py`.
+
+### Write Testable Conformance Tests
+
+Software design fills complete book shelves and this section can only cover some
+basic best practices.
+As a general rule of thumb, the earlier you begin writing unit tests, the better
+– it will force you to write modules that are easy to test.
+
+Divide conformance tests scripts into smaller, loosely coupled units, i.e.,
+functions that serve one purpose.
+Pass externally retrieved data as function arguments instead of hardcoding the
+calls to the retrieval functions inside a function.
```

Comment:
Prefix this by saying: `Practice inversion of control, that is to say, ` (or something similar)